src/core/thread_pool.hh

```diff
@@ -38,11 +65,11 @@ public:
     explicit thread_pool(reactor& r, sstring thread_name);
     ~thread_pool();
     template <typename T, typename Func>
-    future<T> submit(Func func) noexcept {
-        ++_aio_threaded_fallbacks;
+    future<T> submit(Func func, submit_reason reason = submit_reason::unknown) noexcept {
```

Comment:
Ah, let's drop the default and adjust all callers.

It's nicer to put the lambda as the last parameter so there's nothing after the lambda body in the call site.

Reply:
Adjusted the argument order to make `submit_reason` first. Dropped the default value for the parameter.